### This jupyter notebook contains contents of the lesson "Applying functions, sorting and ranking in the datasets"

**Author : Umidjon Sattorov. Machine Learning engineer**

In [24]:
# Importing necessary libraries and loading required dataframes
import pandas as pd
import numpy as np 

# Loading dataframes
df_cardio = pd.read_csv('./datasets/medical_examination.csv')
df_tit = pd.read_csv('./datasets/titanic.csv')

df_cardio

id    age  sex  height  weight  ap_hi  ap_lo  cholesterol  gluc  \
0          0  18393    2     168    62.0    110     80            1     1   
1          1  20228    1     156    85.0    140     90            3     1   
2          2  18857    1     165    64.0    130     70            3     1   
3          3  17623    2     169    82.0    150    100            1     1   
4          4  17474    1     156    56.0    100     60            1     1   
...      ...    ...  ...     ...     ...    ...    ...          ...   ...   
69995  99993  19240    2     168    76.0    120     80            1     1   
69996  99995  22601    1     158   126.0    140     90            2     2   
69997  99996  19066    2     183   105.0    180     90            3     1   
69998  99998  22431    1     163    72.0    135     80            1     2   
69999  99999  20540    1     170    72.0    120     80            2     1   

       smoke  alco  active  cardio  
0          0     0       1       0  
1          0     0       1       1  
2          0     0       0       1  
3          0     0       1       1  
4          0     0       0       0  
...      ...   ...     ...     ...  
69995      1     0       1       0  
69996      0     0       1       1  
69997      0     1       0       1  
69998      0     0       0       1  
69999      0     0       1       0  

[70000 rows x 13 columns]

In [20]:
df_tit

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

In [12]:
# Converting age given in days to years
df_cardio['age'] = df_cardio['age'].apply(lambda x : x / 365)
df_cardio

id        age  sex  height  weight  ap_hi  ap_lo  cholesterol  gluc  \
0          0  50.391781    2     168    62.0    110     80            1     1   
1          1  55.419178    1     156    85.0    140     90            3     1   
2          2  51.663014    1     165    64.0    130     70            3     1   
3          3  48.282192    2     169    82.0    150    100            1     1   
4          4  47.873973    1     156    56.0    100     60            1     1   
...      ...        ...  ...     ...     ...    ...    ...          ...   ...   
69995  99993  52.712329    2     168    76.0    120     80            1     1   
69996  99995  61.920548    1     158   126.0    140     90            2     2   
69997  99996  52.235616    2     183   105.0    180     90            3     1   
69998  99998  61.454795    1     163    72.0    135     80            1     2   
69999  99999  56.273973    1     170    72.0    120     80            2     1   

       smoke  alco  active  cardio  
0          0     0       1       0  
1          0     0       1       1  
2          0     0       0       1  
3          0     0       1       1  
4          0     0       0       0  
...      ...   ...     ...     ...  
69995      1     0       1       0  
69996      0     0       1       1  
69997      0     1       0       1  
69998      0     0       0       1  
69999      0     0       1       0  

[70000 rows x 13 columns]

In [18]:
# Converting sex column to numerical values using function
def conv(sex) :
    if sex == 'male' : 
        return 1
    else : 
        return 0
    
df_tit['Sex'] = df_tit['Sex'].apply(conv)
df_tit.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name  Sex   Age  SibSp  Parch  \
0                            Braund, Mr. Owen Harris    0  22.0      1      0   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...    0  38.0      1      0   
2                             Heikkinen, Miss. Laina    0  26.0      0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)    0  35.0      1      0   
4                           Allen, Mr. William Henry    0  35.0      0      0   

             Ticket     Fare Cabin Embarked  
0         A/5 21171   7.2500   NaN        S  
1          PC 17599  71.2833   C85        C  
2  STON/O2. 3101282   7.9250   NaN        S  
3            113803  53.1000  C123        S  
4            373450   8.0500   NaN        S

In [25]:
# Extracting first word from the Name column
df_tit['first_name'] = df_tit['Name'].apply(lambda x : x.split()[0])
df_tit.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  first_name  
0      0         A/5 21171   7.2500   NaN        S     Braund,  
1      0          PC 17599  71.2833   C85        C    Cumings,  
2      0  STON/O2. 3101282   7.9250   NaN        S  Heikkinen,  
3      0            113803  53.1000  C123        S   Futrelle,  
4      0            373450   8.0500   NaN        S      Allen,

**Tasks**

1. Create column BMI for the first dataset. BMI stands for Body-Mass-Index which can be measure of how fat person is. If it is higher than 25, then they are considered fat. Formula for BMI :

BMI = mass / (height) ** 2

In [ ]:
# Your code goes here


2) Instead of columns for blood pressure, create column high_pres for indication of person which has high blood pressure. If person's ap-hi > 140 and ap-lo > 90, they should be considered the person who has high blood pressure.

In [ ]:
# Your code goes here


3 If there is not any indication of person's sex in the second dataset, we can extract this information from the Name column. Just think about it, and try to extract this feature.

In [ ]:
# Your code goes here
